source/ai-server-logger.js

```diff
@@ -0,0 +1,122 @@
+(function() {
+    'use strict';
+
+    angular.module('ai.public.serverLogger', ['ng'])
+        .config(function($provide) {
+            $provide.decorator('$log', function($delegate, $injector) {
+                var logger = $injector.get('AiServerLogger');
+                var delegateError = $delegate.error;
+                var delegateWarn = $delegate.warn;
+                var delegateInfo = $delegate.info;
+                var delegateDebug = $delegate.debug;
+
+                var enhanceLoggingFunction = function(loggingFunction, severity) {
+                    return function() {
+                        var args = [].slice.call(arguments);
```

Comment:
better to use _.toArray .. this will allow V8 optimizations. See https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments 

Is the reason for converting it to an Array because of JSON.stringify? Why bother converting it to an array at all?